### ISSUE TITLE
score_to_hash: Assertion `abs(score) < MATE' failed

### ISSUE THREAD

**Issue by lucasart** [lucasart/Demolito#84]
Minimal reproduction steps (added root move display):
```
ucinewgame
position fen 8/8/5p2/p7/6P1/PP4Kp/2k4P/3r4 b - - 1 moves d1g1 g3h3 c2b3 g4g5
go infinite
[...]
info depth 1 score cp 563 time 0 nodes 26 pv g1g5
root move: g1g5
root move: g1g5
root move: g1g5
root move: g1g5
root move: g1g5
root move: b3a3
root move: f6g5
master.d: ./src/htable.c:23: score_to_hash: Assertion `abs(score) < MATE' failed.
```

**Comment by lucasart** [comment]
Shorter repro:
```
ucinewgame
position fen 8/3r4/5p2/p7/6P1/PP3K1p/2k4P/3N4 b - - 0 64 moves d7d1
go infinite
[...]
info depth 10 score cp -528 time 28 nodes 28857 pv b3b4 d1d3 f3e4 d3a3 b4a5 a3a5 e4f3 a5a4 f3g3 a4a3 g3f4 c2d3 f4f5 a3a4
```

**Comment by lucasart** [comment]
Found the problem. All moves were being pruned, which lead to `bestScore = -INF`. The thing is that moveCount is updated before pruning, because we don't want to return a mate or draw score in this case. It's really not clear what to return in this case, and to avoid polluting the hash table with crap entries (score = -INF, no move), it's safer to not prune the first move. This guarantees that bestScore and bestMove are always updated at least once.

The fix is a functional change, and is likely to increase the size of the search tree. So testing for non-regression sprt(-4,0)...

**Comment by lucasart** [comment]
fix failed.

another solution is to use -MATE as starting value instead of -INF. but then bestScore will start larger than alpha = -INF at root in some cases. so may as well get rid of INF and use MATE instead.

**Comment by lucasart** [comment]
trying another fix first: return alpha when `score = -INF`, skipping TT storage. If that fails, go with the bestScore = -MATE solution.

**Comment by lucasart** [comment]
fixed by 48e9b933ce65fb4f9d970c30f6845e814a059ee1
